### PR TITLE
Drum loop

### PR DESCRIPTION
**Pull request by Sevetro**

### CODE REVIEW

File: src/components/DrumTile.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export default class DrumTile extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+          nextDrumTileId: this.props.nextDrumTileId            
+        }
+    }
+    
+    render() {
+        console.log("Hejka tu DrumTile render, moje id: " + this.props.id + ", a nextDrumTileId to: " + this.props.nextDrumTileId)
```
Comment (mikolajpiotrowicz) [inline src/components/DrumTile.js:12]: Użyj destrukturyzacji aby wyciągnąć zmienne z props i state np. taki kod
`backgroundColor: this.state.nextDrumTileId === this.props.id ? "red" : "white" `
wyglądało by to tak kiedy zdestrukturyzowałbyś te zmienne 
`const { id } = this.props;
const { nextDrumTileId } = this.state;
backgroundColor: nextDrumTileId === id ? "red" : "white" `

File: src/components/DrumLoop.js [hunk 2]
```diff
@@ -1,12 +1,54 @@
...
+                    <button onClick={this.removeTile}>-</button>
+                    <button onClick={this.addTile}>+</button>
+                </div>
+
+                {this.state.drumTileArray}
+            </div>
+        )
+
+    }
+}
```
Comment (mikolajpiotrowicz) [inline src/components/DrumLoop.js:54]: changeMetronomeState = (val) => {
        const { isActive} = this.prop;
        this.props.changeMetronomeState(val, isActive, this.toggleMetronome)
    }
Do metronome JS button onClick

File: src/components/DrumTile.js (hunk 1 above)
Comment (mikolajpiotrowicz) [reply src/components/DrumTile.js:12]: renderTile = (id) => (<DrumTile id={id} key={id} isActive={id === this.props.nextDrumTileId} />)

File: src/components/DrumLoop.js [hunk 3]
```diff
@@ -1,12 +1,54 @@
-import React from 'react'
+import React, { Component } from 'react'
+import DrumTile from './DrumTile';
 
-function DrumLoop(){
+export default class DrumLoop extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
```
[the remark this answers is not in the document]
Comment (Sevetro) [reply src/components/DrumLoop.js:7]: done

File: src/components/DrumTile.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React, { Component } from 'react'
+
+export default class DrumTile extends Component {
+    constructor(props) {
```
[the remark this answers is not in the document]
Comment (Sevetro) [reply src/components/DrumTile.js:4]: done

File: src/App.js [hunk 5]
```diff
@@ -1,38 +1,77 @@
...
+
+    this.flashDrumTile()
   }
 
 
-  changeActive = (toggleMetronome) => {
-    this.setState({ bpmNumber: this.state.bpmNumber, isActive: !this.state.isActive }, toggleMetronome)
+  flashDrumTile = () => {
+
+    console.log("nextDrumTileId before setState : " + this.state.nextDrumTileId)
```
[the remark this answers is not in the document]
Comment (Sevetro) [reply src/App.js:40]: done

File: src/components/DrumLoop.js [hunk 6]
```diff
@@ -1,12 +1,54 @@
...
-)
+    
 
-}
-export default DrumLoop;
\ No newline at end of file
+    addTile = () => {
+        this.setState(prevState => {
+            return {
+                drumTileArray: [
```
[the remark this answers is not in the document]
Comment (Sevetro) [reply src/components/DrumLoop.js:22]: done

File: src/components/DrumLoop.js (hunk 2 above)
Comment (Sevetro) [reply src/components/DrumLoop.js:54]: done

**Pull request closed without merging**